lib/entry.js

```diff
@@ -98,21 +100,39 @@ function SalesforceEntity(callScope) {
     });
     conn.on('refresh', (accessToken, res) => {
       console.log('Keys were updated, res=%j', res);
-      emitKeys({oauth: res});
+      emitKeys({ oauth: res });
     });
 
-    await conn.sobject(cfg.object)
+    const maxFetch = cfg.sizeOfPollingPage || 1000;
+    const singlePagePerInterval = (cfg.singlePagePerInterval !== 'no') ;
+    const maxTime = cfg.endTime ? ` AND LastModifiedDate < ${cfg.endTime}` : '';
+    let hasMorePages = true;
+    let lastSeenTime = snapshot.previousLastModified;
+
+    do {
+      const whereCondition = `LastModifiedDate > ${lastSeenTime}${maxTime}`;
+      await conn.sobject(cfg.object)
         .select('*')
-        .where(`LastModifiedDate > ${snapshot}`)
-        .execute({ autoFetch : true, maxFetch : 10000 }, processResults);
+        .where(whereCondition)
+        .sort({ LastModifiedDate: 1 })
+        .execute({ autoFetch: true, maxFetch }, processResults);
```

Comment:
I'm not sure if this query logic correctly supports the case when the two items split by the page break have the same timestamp.

Reply:
What are your suggestions?

Reply:
The pseudo-code in the standard behavior document lays out an algorithm where you set the time range that is searched across once and then you walk through all pages within that time range.  This way, even if you have say 20 001 records with the same timestamp, you are guarenteed to emit each item at least once.

Reply:
What about items, that could be changed before the starting next execution?
E.g.:
1) The first execution of the trigger was fetched the first page of results
2) One object from the first page was updated before the second execution
3) The second execution of the trigger was fetched the second page of results but one object was skipped because the oldest object is on the first page now. 

The question is: which of the case: your or mine, is meet more often and which one we should handle?

Reply:
Good point.  I guess I (and the OIH standards) didn't consider that case.  I suppose that there are ways that the OIH algortim can be tweaked to handle both cases.  In the mean time, we can leave it how you have it now.

Reply:
Ok.